### PR TITLE
Bench2: Improved Log Integration and Sane JSON Decimal Precision

### PR DESCRIPTION
**Pull request by simpsont-oci**

### CODE REVIEW

File: performance-tests/bench_2/test_controller/ScenarioManager.cpp [hunk 1]
```diff
@@ -354,6 +354,10 @@ void ScenarioManager::execute(const Bench::TestController::AllocatedScenario& al
             if (json_2_idl(ss, report)) {
```
Comment (sonndinh) [inline performance-tests/bench_2/test_controller/ScenarioManager.cpp:354]: This `report` variable has the same name as the last parameter of this function. It's not a problem but could it be renamed to improve readability?

File: performance-tests/bench_2/test_controller/main.cpp [hunk 2]
```diff
@@ -383,14 +387,16 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
       std::cout << "Saving Scenario Allocation to File..." << std::endl;
       std::ofstream file(preallocated_scenario_output_path);
       if (file.is_open()) {
-        if (!idl_2_json(allocated_scenario, file, pretty)) {
+        if (pretty && !idl_2_pretty_json(allocated_scenario, file)) {
+          throw std::runtime_error("Could not encode allocated scenario");
+        } else if (!pretty && !idl_2_json(allocated_scenario, file)) {
           throw std::runtime_error("Could not encode allocated scenario");
         }
```
Comment (sonndinh) [inline performance-tests/bench_2/test_controller/main.cpp:394]:
```suggestion
        if ((pretty && !idl_2_pretty_json(allocated_scenario, file)) ||
            (!pretty && !idl_2_json(allocated_scenario, file))) {
          throw std::runtime_error("Could not encode allocated scenario");
        }
```

File: performance-tests/bench_2/test_controller/main.cpp [hunk 3]
```diff
@@ -444,6 +450,14 @@ int ACE_TMAIN(int argc, ACE_TCHAR* argv[])
       size_t total_worker_reports = 0;
       for (CORBA::ULong i = 0; i < report.node_reports.length(); ++i) {
         total_worker_reports += report.node_reports[i].worker_reports.length();
+        if (show_worker_logs) {
+          for (CORBA::ULong j = 0; j < report.node_reports[i].worker_logs.length(); ++j) {
+            result_file << "=== Showing Log for Node " << report.node_reports[i].node_id << " Worker #" << report.node_reports[i].worker_ids[j] << " ===" << std::endl;
+            result_file << report.node_reports[i].worker_logs[j] << std::endl << std::endl;
+            std::cout << "=== Showing Log for Node " << report.node_reports[i].node_id << " Worker #" << report.node_reports[i].worker_ids[j] << " ===" << std::endl;
+            std::cout << report.node_reports[i].worker_logs[j] << std::endl << std::endl;
```
Comment (sonndinh) [inline performance-tests/bench_2/test_controller/main.cpp:458]:
```suggestion
            std::stringstream header;
            header << "=== Showing Log for Node " << report.node_reports[i].node_id << " Worker #" << report.node_reports[i].worker_ids[j] << " ===" << std::endl;
            result_file << header.str();
            result_file << report.node_reports[i].worker_logs[j] << std::endl << std::endl;
            std::cout << header.str();
            std::cout << report.node_reports[i].worker_logs[j] << std::endl << std::endl;
```

**Review by sonndinh — APPROVED** [review]

File: performance-tests/bench_2/common/json_conversion.h [hunk 4]
```diff
@@ -23,18 +23,34 @@ bool json_2_idl(std::istream& is, IDL_Type& idl_value) {
 }
 
 template<typename IDL_Type>
-bool idl_2_json(const IDL_Type& idl_value, std::ostream& os, bool pretty = false) {
+bool idl_2_json(const IDL_Type& idl_value, std::ostream& os,
```
Comment (iguessthislldo) [inline performance-tests/bench_2/common/json_conversion.h:26]: You could make this a template on `Writer` vs `PrettyWriter` to avoid duplicating the code.

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**